### PR TITLE
chore: fix some bugs about monaco editor crashing

### PR DESCRIPTION
**Pull request by Asalem1**
This was some random attempt to address some latency issues that currently exist on flows that are pretty impossible to recreate locally. It also addresses an issues that crops up in the monaco editor experience when we update the component and try to edit the raw flux. The app crashes, it shouldn't do that, so we're plopping in an ErrorBoundary

### CODE REVIEW

File: src/flows/context/flow.list.tsx [hunk 1]
```diff
@@ -252,20 +252,24 @@ export const FlowListProvider: FC = ({children}) => {
     })
   }
 
-  const update = (id: string, flow: Flow) => {
-    if (!flows.hasOwnProperty(id)) {
-      throw new Error(`${PROJECT_NAME} not found`)
-    }
+  const update = useCallback(
+    (id: string, flow: Flow) => {
+      console.log('UPDATE')
```
Comment (genehynson) [inline src/flows/context/flow.list.tsx:257]: remove this :)

Comment (Asalem1) [reply src/flows/context/flow.list.tsx:257]: 😅

**Review by genehynson — APPROVED** [review]

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 55 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2249)

File: src/flows/context/flow.list.tsx [hunk 2]
```diff
@@ -252,20 +252,23 @@ export const FlowListProvider: FC = ({children}) => {
...
+      }
 
-    setFlows({
-      ...flows,
-      [id]: flow,
-    })
+      setFlows(prevFlows => ({
+        ...prevFlows,
+        [id]: flow,
+      }))
```
Comment (Asalem1) [inline src/flows/context/flow.list.tsx:264]: @drdelambre

Comment (drdelambre) [reply src/flows/context/flow.list.tsx:264]: you beautiful beast you

File: src/shared/components/FluxMonacoEditor.tsx [hunk 3]
```diff
@@ -117,29 +118,31 @@ const FluxEditorMonaco: FC<Props> = ({
...
-          },
-          overviewRulerBorder: false,
-          automaticLayout: true,
-          readOnly: readOnly || false,
-          wordWrap: wrapLines ?? 'off',
-        }}
-        editorDidMount={editorDidMount}
-      />
-    </div>
+    <ErrorBoundary>
```
Comment (drdelambre) [inline src/shared/components/FluxMonacoEditor.tsx:121]: :fire:

**Review by drdelambre — APPROVED** [review]

**Pull request merged**